frontend/packages/knative-plugin/src/components/add/SecretKeySelector.tsx

```diff
@@ -29,7 +29,13 @@ const SecretKeySelector: React.FC<SecretKeySelectorProps & StateProps> = ({
   const [secrets, setSecrets] = React.useState({});
   const fieldId = getFieldId(name, 'secret-key-input');
   const isValid = !(touched && error);
-  const errorMessage = !isValid ? error : '';
+  const errObj = (error as unknown) as { [key: string]: string };
```

Comment:
Why are you typecasting as `unknown`? You can just check the if type is `object` or `string`and then create `errorMessage` based on that.